apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/RetryHandler.cs

```diff
@@ -37,30 +37,35 @@ internal sealed class RetryHandler : IHttpUnsuccessfulResponseHandler
         internal const string InvocationIdHeaderPart = "gccl-invocation-id";
         internal const string AttemptCountHeaderPart = "gccl-attempt-count";
 
-        private static readonly int[] s_retriableErrorCodes =
+        private readonly double _backoffMultiplier;
+        private readonly TimeSpan _initialBackoff;
+        private readonly TimeSpan _maxBackoff;
+        private readonly RetryPredicate _retryPredicate;
+
+        internal RetryHandler(RetryOptions retryOptions)
         {
-                408, // Request timeout
-                429, // Too many requests
-                500, // Internal server error
-                502, // Bad gateway
-                503, // Service unavailable
-                504 // Gateway timeout
-        };
-        private static readonly RetryHandler s_instance = new RetryHandler();
-
-        private RetryHandler() { }
-
-        internal static void MarkAsRetriable<TResponse>(StorageBaseServiceRequest<TResponse> request)
+            _retryPredicate = retryOptions.RetryPredicate;
+            _backoffMultiplier = retryOptions.RetryTimings.BackoffMultiplier;
+            _initialBackoff = retryOptions.RetryTimings.InitialBackoff;
+            _maxBackoff = retryOptions.RetryTimings.MaxBackoff;
+        }
+
+        // TODO: Provided null options for now, will remove it once the implementation across all APIs is complete
```

Comment:
I'd suggest having a separate overload without the RetryOptions, that we intend to remove at the end. Make `options` a required parameter here, and define what a value of null means. (I'd suggest either "never" or prohibit it entirely.)